tests/backends/base/test_creation.py

```diff
@@ -40,3 +41,29 @@ def test_custom_test_name_with_test_prefix(self):
         test_connection.settings_dict['TEST'] = {'NAME': test_name}
         signature = BaseDatabaseCreation(test_connection).test_db_signature()
         self.assertEqual(signature[3], test_name)
+
+
+@mock.patch.object(connection, 'ensure_connection', return_value=None)
+@mock.patch('django.core.management.commands.migrate.Command.handle', return_value=None)
+class TestDbCreationTests(SimpleTestCase):
+    def test_migrate_test_setting_false(self, mocked_migrate, mocked_ensure_connection):
+        creation = connection.creation_class(connection)
+        try:
+            saved_settings = connection.settings_dict
+            connection.settings_dict['TEST']['MIGRATE'] = False
+            with mock.patch.object(creation, '_create_test_db', return_value=None):
+                creation.create_test_db(verbosity=0, autoclobber=True, serialize=False)
+            mocked_migrate.assert_not_called()
+        finally:
+            connection.settings_dict = saved_settings
```

Comment:
This does not restore the original settings. The settings are being mutated in place so `connection.settings_dict` and `saved_settings` reference the same dict, not a copy.

I tested this by placing `print()` calls after the `finally` block.

Was the intent to create a deep copy first?